net/scan.h

```diff
@@ -0,0 +1,62 @@
+#ifndef __SCAN_H
+#define __SCAN_H
+
+/**
+ * @file
+ * @ingroup     net_scan
+ *
+ * @brief       Scan list implementation
+ *
+ * @author Geovane Fedrecheski <geovane.fedrecheski@inria.fr>
+ *
+ * @copyright Inria, 2024
+ */
+
+#include <nrf.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "device.h"
+#include "protocol.h"
+#include "scheduler.h"
+
+//=========================== defines =========================================
+#define BLINK_MAX_SCAN_LIST_SIZE (10)
+#define BLINK_SCAN_OLD_US (1000*1000*3) // rssi reading considered old after 3 seconds
+#define BLINK_SCAN_HANDOVER_HYSTERESIS (6) // hysteresis (in dBm) for handover
+
+//=========================== variables =======================================
+typedef struct {
+    int8_t rssi;
+    uint32_t timestamp;
+} dl_rssi_t;
+
+typedef struct {
+    uint64_t gateway_id;
+    dl_rssi_t rssi[DOTLINK_N_BLE_ADVERTISING_FREQUENCIES]; // channels 37, 38, 39
+} dl_scan_t;
+
+//=========================== prototypes ======================================
+
+/**
+ * Adds a new rssi reading
+ *
+ * @param gateway_id the gateway id
+ * @param rssi the rssi reading in dBm
```

Comment:
ideally, your naming is to clear that one understands what the function does. I recommend to only add this is very very obscrure.

Reply:
addressed